### PR TITLE
Limit CopyLinkWithStrictCleaning by http(s) urls

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/26236

- Hiding item for non-http links, can be checked on the page https://fmarier.github.io/brave-testing/copy-clean-link.html w

image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Use https://fmarier.github.io/brave-testing/copy-clean-link.html

### CODE REVIEW

**Review by goodov — APPROVED** [review]

File: chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc [hunk 1]
```diff
@@ -401,7 +401,7 @@ void BraveRenderViewContextMenu::InitMenu() {
                : IDS_CONTENT_CONTEXT_OPENLINKTOR);
   }
 #endif
-  if (!params_.link_url.is_empty()) {
+  if (!params_.link_url.is_empty() && params_.link_url.SchemeIsHTTPOrHTTPS()) {
```
[the remark this answers is not in the document]
Comment (fmarier) [reply chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc:404]: I've added a new section on https://fmarier.github.io/brave-testing/copy-clean-link.html to test this.

[the remark this answers is not in the document]
Comment (spylogsster) [reply chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc:404]: this covers only renderer context menu

[the remark this answers is not in the document]
Comment (spylogsster) [reply chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc:404]: image

**Review by fmarier — CHANGES_REQUESTED** [review]
> this covers only renderer context menu

If it's easy to do in the other context menus, it would be good to do it there too.

Also, I'd suggest putting an early return in `SanitizeURL()` if a URL with a non-HTTP / non-HTTPS scheme is encountered. In these cases, just returning the original URL.

**Comment by spylogsster** [comment]
> > this covers only renderer context menu
> 
> If it's easy to do in the other context menus, it would be good to do it there too.
> 
> Also, I'd suggest putting an early return in `SanitizeURL()` if a URL with a non-HTTP / non-HTTPS scheme is encountered. In these cases, just returning the original URL.

done

**Review by fmarier — APPROVED** [review]
Thanks for the comprehensive change and the tests!

**Pull request merged**